vissl/engines/benchmark_suite_scheduler.py

```diff
@@ -0,0 +1,545 @@
+                                retry all failed jobs.
+        max_retries: (int, optional, default=3). Maximum number of retries.
+        """
+        self.evaluation_jobs_finished = set()
+
+        # Required Arguments
+        self.training_checkpoint_dir = training_checkpoint_dir
+        self.training_checkpoint_file = os.path.join(
+            self.training_checkpoint_dir, "train_config.yaml"
+        )
+        self.benchmarks = benchmarks
+
+        # Optional Arguments
+        self.evaluate_final_phase = evaluate_final_phase
+        self.evaluation_phase_freq = evaluation_phase_freq
+        self.evaluation_iter_freq = evaluation_iter_freq
+        self.max_training_iterations = max_training_iterations
+        self.autoload_slurm_evaluator_checkpoint = autoload_slurm_evaluator_checkpoint
+        self.slurm_evaluator_checkpoint = slurm_evaluator_checkpoint
+        self.retry_evaluation_job_ids = retry_evaluation_job_ids or []
+        self.auto_retry_evaluations = auto_retry_evaluations
+        self.max_retries = max_retries
+
+        self._validate_class_instance()
+
+        # Will be set in #evaluate, once training_checkpoint_dir becomes available.
+        self.training_config = None
+        self.evaluation_results = None
+
+    def evaluate(self):
+        """
+        Evaluate the checkpoints. At a high level, this is the structure.
+
+        1. Load training YAML config file.
+        2. Monitor training checkpoints. When checkpoint is ready, launch the evaluation jobs.
+        3. Monitor the evaluation jobs. When evaluation jobs are complete, mark the results.
+        3. Once all jobs have been recorded, complete.
+        """
+        start_time = time.time()
+
+        # Wait for the training checkpoint folder to be available.
+        while True:
+            if time.time() - start_time > _TRAINING_CHECKPOINT_FILE_WAIT_SECONDS:
+                raise RuntimeError(
+                    f"Training checkpoint folder still doesn't exist after:"
+                    f"{_TRAINING_CHECKPOINT_FILE_WAIT_SECONDS / 60} minutes"
+                )
+
+            if PathManager.exists(self.training_checkpoint_file):
+                # Load training yaml config.
+                self._load_training_config()
+                break
+
+            time.sleep(_SLEEP_TIME_SECONDS)
+
+        self.save_evaluation_results()
+
+        # Checkpoint folder is now available. Continuously monitor the training checkpoints,
+        # launch evaluation jobs as needed, monitor their progress, and record their results.
+        while True:
+            self._evaluate_checkpoints()
+
+            self._check_evaluation_jobs()
+
+            # Break if no more checkpoints to evaluate
+            if self._finished():
+                print("Evaluations are finished")
+                break
+
+            time.sleep(_SLEEP_TIME_SECONDS)
+
+    def save_evaluation_results(self):
```

Comment:
nit: would it make sense to rename it something like "setup_evaluation_out_folders" etc something like this i.e use the word `setup` ?